### PR TITLE
Added support for non-NaNish base 16 numbers in jsonpath

### PR DESCRIPTION
**Pull request by lulzmachine**
This is related to https://github.com/deepstreamIO/deepstream.io-client-js/issues/327

### CODE REVIEW

**Comment by yasserf** [comment]
Hey, awesome thank you! Would you mind also signing our CLA? https://deepstream.io/info/community/cla/

File: test-unit/unit/record/json-pathSpec.js [hunk 1]
```diff
@@ -137,6 +137,16 @@ describe( 'objects are created from paths and their value is set correctly', fun
...
 	});
 
+	it( 'even when the path is not NaNish and could be interpreted as a base 16 number', function() {
+		var record = {};
+    let pathName = '0x02335';
+    record = jsonPath.set( record, pathName, 'value' );
+    expect( record[ 0 ] ).toBe( undefined );
+		expect( record[ pathName] ).toBe( 'value' );
+    expect( jsonPath.get( record, pathName )).toBe( 'value' );
+    expect( record[ pathName ]).toBe( 'value' );
```
Comment (ralphtheninja) [inline test-unit/unit/record/json-pathSpec.js]: Maybe adjust the tabs here :)

File: src/record/json-path.js [hunk 2]
```diff
@@ -124,6 +124,6 @@ function tokenize( path ) {
 	}
 
 	return cache[ path ] = parts.map( function( part ) {
-		return !isNaN( part ) ? parseInt( part, 10 ) : part;
+		return part;
```
Comment (ralphtheninja) [inline src/record/json-path.js]: Maybe do

```js
return cache[ path ] = parts.slice(0);
```

Comment (lulzmachine) [inline src/record/json-path.js]: Hm actually, parts is created just a few lines above it, and isn't sent anywhere. So I guess I don't even need to copy it. I'll just `cache[ path ] = parts`

Comment (ralphtheninja) [inline src/record/json-path.js]: > Hm actually, parts is created just a few lines above it, and isn't sent anywhere. So I guess I don't even need to copy it. I'll just cache[ path ] = parts

Even better :)

File: test-unit/unit/record/json-pathSpec.js [hunk 3]
```diff
@@ -137,6 +137,16 @@ describe( 'objects are created from paths and their value is set correctly', fun
 		});
 	});
 
+	it( 'even when the path is not NaNish and could be interpreted as a base 16 number', function() {
+		var record = {};
+		let pathName = '0x02335';
```
Comment (yasserf) [inline test-unit/unit/record/json-pathSpec.js]: let fails build ( we can put a use strict at the top since this just tests though if you want )

Comment (lulzmachine) [inline test-unit/unit/record/json-pathSpec.js]: Ok. I changed to "var". Thanks!

**Comment by yasserf** [comment]
Great thanks!

**Comment by ralphtheninja** [comment]
@yasserf How often do you release versions? We could really use an updated version of the client lib. I'm not sure what to bump here, maybe it should be `npm version major` since this PR affects the key format.

**Comment by yasserf** [comment]
Hey @ralphtheninja, can do a quick release with that. Don't think this should be a major though, otherwise we will need to postpone it till 3.0 which is not due for a while.

**Comment by ralphtheninja** [comment]
> Hey @ralphtheninja, can do a quick release with that. Don't think this should be a major though, otherwise we will need to postpone it till 3.0 which is not due for a while.

Check! A patch can be fine too. It depends on how you look at the change, since you could consider this a bug fix :)

**Comment by yasserf** [comment]
awesome, will plan a release to coincide with deepstream.io asap, just pending changelogs

**Pull request merged**